src/hybrid-appliance/azext_hybrid_appliance/custom.py

```diff
@@ -0,0 +1,233 @@
+    kubeconfig_path = utils.get_kubeconfig_path()
+    kubectl_client_location = utils.install_kubectl_client()
+    latestMajorVersion, latestMinorVersion = utils.get_latest_tested_microk8s_version()
+    os.environ["MICROK8S_VERSION"] = "{}.{}".format(latestMajorVersion, latestMinorVersion)
+    os.environ["KUBECTL_CLIENT_LOCATION"] = "{}".format(kubectl_client_location)
+    os.environ["KUBECONFIG_PATH"]=kubeconfig_path
+
+    current_path = os.path.abspath(os.path.dirname(__file__))
+    script_file_path = os.path.join(current_path, "microk8sbootstrap.sh")
+    cmd = ["sh", script_file_path]
+
+    process = subprocess.Popen(cmd)
+    if process.wait() != 0:
+        telemetry.set_exception(exception="Microk8s cluster setup failed", fault_type=consts.MicroK8s_Setup_Failed, summary="Microk8s cluster provisioning failed")
+        raise CLIInternalError("Failed to setup microk8s cluster")
+    
+    if not utils.check_microk8s():
+        telemetry.set_exception(exception="Microk8s cluster is not running", fault_type=consts.MicroK8s_Cluster_Not_Running, summary="Microk8s cluster is not running after the set up")
+        raise CLIInternalError("Microk8s cluster is not running after setting up the cluster. Please check the logs tarball at /var/snap/microk8s/current")
+
+    # Onboard the cluster to arc
+    cmd_onboard_arc= ['connectedk8s', 'connect', '-n', name, '-g', resource_group_name, '--kube-config', kubeconfig_path]
+
+    # The NO_PROXY env variable needs the api server address, which is not available with the user before the cluster is provisioned.
+    # To get around that, we get the server address and services cidr and append the no_proxy variable with those values.
+    if http_proxy or https_proxy or no_proxy:
+        no_proxy = utils.get_proxy_parameters(no_proxy)
+        os.environ["NO_PROXY"] = no_proxy
+    if location:
+        cmd_onboard_arc.extend(["--location", location])
+    if http_proxy:
+        cmd_onboard_arc.extend(["--proxy-http", http_proxy])
+    if https_proxy:
+        cmd_onboard_arc.extend(["--proxy-https", https_proxy])
+    if no_proxy:
+        cmd_onboard_arc.extend(["--proxy-skip-range", no_proxy])
+    if proxy_cert:
+        cmd_onboard_arc.extend(["--proxy-cert", proxy_cert])
+
+    onboarding_result = get_default_cli().invoke(cmd_onboard_arc)
+    if onboarding_result != 0:
+        telemetry.set_exception(exception="Arc onboarding of the k8s cluster failed", fault_type=consts.Arc_Onboarding_Failed, summary="Arc onboarding of the k8s cluster failed")
+        raise CLIInternalError("Onboarding the k8s cluster to Arc failed")
+    else:
+        print("The k8s cluster has been onboarded to Arc successfully")
+
+def upgrade_hybrid_appliance(resource_group_name, name):
+    try:
+        azure_clusterconfig_cm = subprocess.check_output(['microk8s', 'kubectl', 'get', 'cm', 'azure-clusterconfig', '-n', 'azure-arc', '-o', 'json']).decode()
```

Comment:
we can use python SDK https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#read_namespaced_config_map